PastaGA.py

```diff
@@ -0,0 +1,354 @@
+        parentchance_2 = random.random()
+        prob_dic1 = self.__prob_dict
+        prob_dic2 = self.__prob_dict
+        def_check = 0
+        for chrom in prob_dic1:
+            if prob_dic1[chrom] >= parentchance_1:
+                parent_1 = chrom
+                def_check = 1
+            if def_check == 1:
+                break
+        def_check = 0
+        prev_parent = None
+        for chrom in prob_dic2:
+            if prob_dic2[chrom] >= parentchance_2:
+                parent_2 = chrom
+                #if parent_2 == parent_1:
+                    #if prev_parent != None:
+                        #parent_2 = prev_parent
+                    #if prev_parent == None:
+                        #parent_2 = next(chrom)
+                def_check = 1
+            if def_check == 1:
+                break
+            prev_parent = chrom
+        parent_tuple = (parent_1, parent_2)
+        return parent_tuple
+
+    def dict_producer(self):
+        for chrom in self.__chrom_list:
+            chrom.set_fitvalue()
+            self.__chrom_fitdict[chrom] = chrom.get_fitvalue()
+        pass
+
+    def breed_et_mutate(self, couple_tuple):
+        #mutates children after bred from two chromosomes
+        pasta1 = couple_tuple[0]
+        pasta2 = couple_tuple[1]
+        ingredients_1 = pasta1.list_ing()
+        ingredients_2 = pasta2.list_ing()
+        ingredients_3 = []
+        mutated_child = []
+        for num in range(0,1):
+            random_layout = random.randint(1, 6)
+            if random_layout == 1:
+                ingredients_3 = [ingredients_1[0],ingredients_1[1], ingredients_2[2], ingredients_2[3]]
+            if random_layout == 2:
+                ingredients_3 = [ingredients_1[0],ingredients_2[1], ingredients_1[2], ingredients_2[3]]
+            if random_layout == 3:
+                ingredients_3 = [ingredients_1[0],ingredients_2[1], ingredients_2[2], ingredients_1[3]]
+            if random_layout == 4:
+                ingredients_3 = [ingredients_2[0],ingredients_2[1], ingredients_1[2], ingredients_1[3]]
+            if random_layout == 5:
+                ingredients_3 = [ingredients_2[0],ingredients_1[1], ingredients_1[2], ingredients_2[3]]
+            if random_layout == 6:
+                ingredients_3 = [ingredients_2[0],ingredients_1[1], ingredients_2[2], ingredients_1[3]]
+        for ing in ingredients_3:
+            mutate_poss = random.randint(0,100)
+            if mutate_poss < mutate_chance:
+                ing = poss_ingredients[random.randint(0, len(poss_ingredients)-1)]
+            mutated_child.append(ing)
+        child = pasta(mutated_child[0], mutated_child[1], mutated_child[2], mutated_child[3])
+        self.__chrom_list.append(child)
+
+    def reset_population(self):
+        #replaces current dict with new generated genes/use to create initial population
+        pasta_count = 0
+        prev_pasta = pasta("spaghetti", "a", "a", "a")
```

Comment:
Forgot to get rid of that, it was being used to check for a bug where some chromosomes were getting overwritten.